app/src/main/java/com/rpfcoding/snoozeloo/feature_alarm/presentation/add_edit/AddEditAlarmScreen.kt

```diff
@@ -0,0 +1,503 @@
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Slider
+import androidx.compose.material3.SliderDefaults
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.rpfcoding.snoozeloo.R
+import com.rpfcoding.snoozeloo.core.components.InputTimeTextField
+import com.rpfcoding.snoozeloo.core.presentation.designsystem.SnoozelooTheme
+import com.rpfcoding.snoozeloo.core.util.showToast
+import com.rpfcoding.snoozeloo.feature_alarm.domain.DayValue
+import com.rpfcoding.snoozeloo.feature_alarm.presentation.components.DayChip
+import kotlinx.coroutines.flow.collectLatest
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun AddEditAlarmScreenRoot(
+    navigateBack: () -> Unit,
+    navigateToRingtoneList: () -> Unit,
+    alarmId: String?,
+    viewModel: AddEditAlarmViewModel = koinViewModel()
+) {
+    val context = LocalContext.current
+    val state = viewModel.state
+    var isNavigatingBack by remember { mutableStateOf(false) }
+
+    BackHandler {}
```

Comment:
Empty backhandler, system back nav won’t work. `BackHandler` is enabled by default when called